src/test/java/software/amazon/awssdk/crt/test/SigningTest.java

```diff
@@ -271,6 +274,53 @@ public Credentials getCredentials() {
         }
         assertTrue(failed);
     }
+    @Test
+    public void testSigningAnonymous() throws Exception {
+            HttpRequest request = createSimpleRequest("https://www.example.com", "POST", "/derp", "<body>Hello</body>");
+            try (AwsSigningConfig config = new AwsSigningConfig()) {
+                config.setAlgorithm(AwsSigningConfig.AwsSigningAlgorithm.SIGV4);
+                config.setSignatureType(AwsSigningConfig.AwsSignatureType.HTTP_REQUEST_VIA_HEADERS);
+                config.setRegion("us-east-1");
+                config.setService("service");
+                config.setTime(System.currentTimeMillis());
+                config.setUseDoubleUriEncode(true);
+                config.setShouldNormalizeUriPath(true);
+                config.setSignedBodyValue(AwsSigningConfig.AwsSignedBodyValue.EMPTY_SHA256);
+                config.setCredentials(Credentials.createAnonymousCredentials());
+
+                CompletableFuture<HttpRequest> result = AwsSigner.signRequest(request, config);
+                HttpRequest signedRequest = result.get();
+                assertNotNull(signedRequest);
+                assertFalse(hasHeader(signedRequest, "Authorization"));
+            }
+    }
+
+    @Test(expected = CrtRuntimeException.class)
+    public void testSigningNullFail() throws Exception {
+        HttpRequest request = createSimpleRequest("https://www.example.com", "POST", "/derp", "<body>Hello</body>");
+        try{
+            AwsSigningConfig config = new AwsSigningConfig();
+            config.setAlgorithm(AwsSigningConfig.AwsSigningAlgorithm.SIGV4);
+            config.setSignatureType(AwsSigningConfig.AwsSignatureType.HTTP_REQUEST_VIA_HEADERS);
+            config.setRegion("us-east-1");
+            config.setService("service");
+            config.setTime(System.currentTimeMillis());
+            config.setUseDoubleUriEncode(true);
+            config.setShouldNormalizeUriPath(true);
+            config.setSignedBodyValue(AwsSigningConfig.AwsSignedBodyValue.EMPTY_SHA256);
+
+            CompletableFuture<HttpRequest> result = AwsSigner.signRequest(request, config);
+            HttpRequest signedRequest = result.get();
+            assertNotNull(signedRequest);
+            assertFalse(hasHeader(signedRequest, "Authorization"));
+        } catch (ExecutionException e){
+            assertEquals(e.getCause().getClass(), CrtRuntimeException.class);
+            CrtRuntimeException crtRuntimeException = (CrtRuntimeException) e.getCause();
+            assertEquals(crtRuntimeException.errorCode, 6149);
```

Comment:
assert on the `.errorName` instead of the `.errorCode`
we realize that CrtRuntimeException isn't very java-esque (it uses error codes instead of subclasses). But checking the integer value is particularly meaningless to someone just reading this code